utilities/cli/holohub.py

```diff
@@ -318,51 +333,58 @@ def get_buildtype_str(self, build_type: Optional[str]) -> str:
+        build_type = self.get_buildtype_str(build_type)
+        build_dir = HoloHubCLI.DEFAULT_BUILD_PARENT_DIR / project_name
+        build_dir.mkdir(parents=True, exist_ok=True)
+
+        holohub_cli_util.run_command(
+            [
+                "cmake",
+                "-B",
+                str(build_dir),
+                "-S",
+                str(HoloHubCLI.HOLOHUB_ROOT),
+                "-G",
+                "Ninja",
+                f"-DCMAKE_BUILD_TYPE={build_type}",
+                f"-DCMAKE_PREFIX_PATH={HoloHubCLI.DEFAULT_SDK_DIR}",
+                f"-DHOLOHUB_DATA_DIR:PATH={HoloHubCLI.DEFAULT_DATA_DIR}",
+                f"-DAPP_{project_name}=ON",
+            ],
+            dry_run=dryrun,
+        )
+        holohub_cli_util.run_command(
+            ["cmake", "--build", str(build_dir), "--config", build_type], dry_run=dryrun
+        )
+
+        return build_dir, project_data
+
     def handle_build(self, args: argparse.Namespace) -> None:
         """Handle build command"""
         if args.local or os.environ.get("HOLOHUB_BUILD_LOCAL"):
-            project_data = self._find_project(
+            self._build_project_locally(
                 project_name=args.project,
                 language=args.language if hasattr(args, "language") else None,
+                build_type=args.build_type,
+                dryrun=args.dryrun,
             )
-            if not project_data:
-                holohub_cli_util.fatal(
-                    f"Project '{args.project}' {f'({args.language})' if hasattr(args, 'language') else ''} not found"
-                )
-
-            build_type = self.get_buildtype_str(args.build_type)
-            build_dir = HoloHubCLI.DEFAULT_BUILD_PARENT_DIR / args.project
-            build_dir.mkdir(parents=True, exist_ok=True)
-
-            holohub_cli_util.run_command(
-                [
-                    "cmake",
-                    "-B",
-                    str(build_dir),
-                    "-S",
-                    str(HoloHubCLI.HOLOHUB_ROOT),
-                    "-G",
-                    "Ninja",
-                    f"-DCMAKE_BUILD_TYPE={build_type}",
-                    f"-DCMAKE_PREFIX_PATH={HoloHubCLI.DEFAULT_SDK_DIR}",
-                    f"-DHOLOHUB_DATA_DIR:PATH={HoloHubCLI.DEFAULT_DATA_DIR}",
-                    f"-DAPP_{args.project}=ON",
-                ],
-                dry_run=args.dryrun,
-            )
-            holohub_cli_util.run_command(
-                ["cmake", "--build", str(build_dir), "--config", build_type], dry_run=args.dryrun
-            )
-
         else:
             # Build in container
             container = self._make_project_container(
                 project_name=args.project,
                 language=args.language if hasattr(args, "language") else None,
+                dryrun=args.dryrun,
+                build_args={},  # Empty dict to trigger build but with no special args
```

Comment:
Concerned that the default behavior here with `None` versus `{}` is less intuitive than calling `.build` explicitly